actionpack/lib/action_controller/metal/request_forgery_protection.rb

```diff
@@ -87,6 +87,7 @@ module ClassMethods
       #
       # * <tt>:only/:except</tt> - Passed to the <tt>before_action</tt> call. Set which actions are verified.
       # * <tt>:if/:unless</tt> - Passed to the <tt>before_action</tt> call. Set when actions are verified.
+      # * <tt>:prepend</tt> - ?
```

Comment:
The two first options should be described in terms of before_action. That’s all internal stuff. Maybe just:

only/except - Only apply forgery protection to a subset of actions. Like only: [ :create, :create_all ]
if/unless - Turn off the forgery protection entirely depending on the passed proc or method reference.

Also, needs to say “You can disable forgery protection” not just CRSF — since we also check same origin.




> On Jan 4, 2015, at 5:46 PM, Josef Šimánek <notifications@github.com> wrote:
> 
> In actionpack/lib/action_controller/metal/request_forgery_protection.rb <https://github.com/rails/rails/pull/18334#discussion-diff-22445355>:
> 
> > @@ -87,6 +87,7 @@ module ClassMethods
> >        #
> >        # * <tt>:only/:except</tt> - Passed to the <tt>before_action</tt> call. Set which actions are verified.
> >        # * <tt>:if/:unless</tt> - Passed to the <tt>before_action</tt> call. Set when actions are verified.
> > +      # * <tt>:prepend</tt> - ?
> Thanks.
> 
>  <https://cloud.githubusercontent.com/assets/193936/5608460/edecd69a-9484-11e4-8602-431ce1246175.png>
> —
> Reply to this email directly or view it on GitHub <https://github.com/rails/rails/pull/18334/files#r22445355>.
>